### ISSUE TITLE
Ordered Bag Implementation?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [NDuma/artemis-framework#2]
```
TL;DR: I want an OrderedBag implementation so that I only have to call 
glUseProgram once per OpenGL ES 2.0 shader.

Hi Artemis devs. I really like the library and it seems very nice so far. I was 
just integrating it with a libgdx game I am making and I was trying to make a 
RenderSystem for objects that use OpenGL ES 2.0. As you may know, it is more 
efficient if you do not repetitively use the glUseProgram method. It is much 
nicer to call glUseProgram once per shader and render everything that uses that 
shader in one hit. Therefore I was wondering if we could possibly add in an 
OrderedBag implementation to Artemis. It may be a little bit slower but it 
would guarantee that I could get speed improvements while the program was 
actually running and when everything was finally in the bag. 
```

Original issue reported on code.google.com by `robertma...@gmail.com` on 12 Aug 2012 at 4:22
